### DISCUSSION TITLE
How to use Docker on Synology for backup Teslamate

### DISCUSSION THREAD

**Discussion by socekin** [teslamate-org/teslamate#3838] [Q&A]
I use Synology's container management to create the Teslamate project, and it runs normally. Since Synology does not support the docker compose command, I can only use the docker-compose command. I use this command to perform a backup:  
```shell
sudo docker-compose exec -T database pg_dump -U teslamate teslamate > /volume1/docker/teslamate.bck
``` 

However, it prompts an error: "service 'database' is not running container #1". anyone knows how to fix it? thanks

![CleanShot 2024-04-14 at 00 28 01@2x](https://github.com/teslamate-org/teslamate/assets/874961/4d3e5031-1927-4466-be80-186d45b05ea6)

**Comment by bsf29** [comment]
Sounds like you stopped the DB when you are attempting to create the backup and it needs to be running.  I'm using DSM's Scheduled Task to run my bash script to take weekly backup.  Prior to running the docker-compose backup command, I restart TeslaMate to make sure everything is up running so the backup completes.  

Oh, just noticed you are missing -d switch: docker-compose exec -T database pg_dump -U teslamate **-d teslamate**

**Reply by socekin** [reply]
I try restart TeslaMate, and teslamate database is running. but still prompts an error: "service 'database' is not running container "
<img width="934" alt="image" src="https://github.com/teslamate-org/teslamate/assets/874961/0be2166e-4e7b-45a8-b59c-39db97a91e38">

**Comment by socekin** [comment]
Finally, i use this command to backup my data. teslamate-database-1 is the container name
`sudo docker exec -it teslamate-database-1 pg_dump -U teslamate teslamate > /volume1/docker/teslamate.bck`

**Reply by satanforaday** [reply]
This worked for me, thanks.
